File: TenMinsPy/day90.py
# Ask for a rowid, delete csv row after confirmation prompt

import csv

def deleteCSVRow(row_id):
    try:
        ans = input("are you sure? (y/n)")
        if ans.strip().lower() == "y":
            # Step 1: Read all lines
            file_path="D:/coding30minsAday/textFiles/JTrialCsv.csv"
            with open(file_path, "r") as f:
                lines = f.readlines()

            # Step 2: Modify the value
            if len(lines) >= row_id:
                lines.pop(row_id - 1)

            # Step 3: Rewrite the CSV
            with open(file_path, "w") as f:
                f.writelines(lines)
        else:
            return

    except Exception as e:
        print("Error:", e)

File: TenMinsJava/day90.java
// Ask for a rowid, delete csv row after confirmation prompt
package TenMinsJava;

import java.io.File;
import java.io.FileWriter;
import java.util.ArrayList;
import java.util.Scanner;

public class day90 {
    public static void deleteCSVRow(int rowId) {
        try {
            Scanner choice = new Scanner(System.in);
            System.out.println("Are you sure? (y/n");
            String ans = choice.nextLine().trim().toLowerCase();
            if (ans.equals("y")) {
                File file = new File("D:/coding30minsAday/textFiles/JTrialCsv.csv");
                Scanner fScan = new Scanner(file);
                ArrayList<String> lines = new ArrayList<>();

                // Step 1: Read all lines
                while (fScan.hasNextLine()) {
                    lines.add(fScan.nextLine());
                }
                fScan.close();

                // modifies the value
                lines.remove(rowId - 1);

                // rewrite the csv
                FileWriter w = new FileWriter(file);
                for (String string : lines) {
                    w.write(string + "\n");
                }
                w.close();
            }
            else {
                return;
            }
            
        } catch (Exception e) {
            System.err.println(e);
        }
    }
}
